src/test/java/com/newrelic/jfr/toevent/JITCompilationMapperTest.java

```diff
@@ -50,4 +48,28 @@ void testApply() {
 
     assertEquals(expected, result);
   }
+
+  @Test
+  void read_real_event() throws Exception {
+    final Path dumpFile = Path.of("src", "test", "resources", "startup3.jfr");
+
+    var mapper = new JITCompilationMapper();
+    try (final var recordingFile = new RecordingFile(dumpFile)) {
+      while (recordingFile.hasMoreEvents()) {
+        var event = recordingFile.readEvent();
+
+        if (event != null) {
+          if (event.getEventType().getName().equals(JITCompilationMapper.EVENT_NAME)) {
+            var le = mapper.apply(event);
+            assertEquals(1, le.size());
+            assertEquals(109L, le.get(0).getAttributes().asMap().get("duration"));
+            assertEquals(
+                "java.lang.invoke.LambdaForm$Name.replaceNames([Ljava/lang/invoke/LambdaForm$Name;[Ljava/lang/invoke/LambdaForm$Name;II)Ljava/lang/invoke/LambdaForm$Name;",
+                le.get(0).getAttributes().asMap().get("desc"));
+            break;
+          }
+        }
+      }
+    }
+  }
```

Comment:
unfortunately, this test still passes if the file has no events or just null events or if none of the events are for the `JITCompilationMapper`.  There are several ways of making this fail under these cases (which is preferred), but the stupidest/simplest way is to set a flag after the assertions and verify it at the end of the test method.

Reply:
Fixed.